90.md

```diff
@@ -0,0 +1,171 @@
+NIP-90
+======
+
+Data Vending Machine
+--------------------
+
+`draft` `optional` `author:pablof7z` `author:dontbelievethehype`
+
+This NIP defines the interaction between customers and Service Providers for performing on-demand computation.
+
+Money in, data out.
+
+## Kinds
+This NIP reserves the range `5000-7000` for data vending machine use.
+
+| Kind | Description |
+| ---- | ----------- |
+| 5000-5999 | Job request kinds |
+| 6000-6999 | Job result |
+| 7000 | Job feedback |
+
+Job results always use a kind number that is `1000` higher than the job request kind. (e.g. request: `kind:5001` gets a result: `kind:6001`).
+
+Job request types are defined [separately](https://github.com/nostr-protocol/data-vending-machines/tree/master/kinds).
+
+## Rationale
+Nostr can act as a marketplace for data processing, where users request jobs to be processed in certain ways (e.g., "speech-to-text", "summarization", etc.), but they don't necessarily care about "who" processes the data.
+
+This NIP is not to be confused with a 1:1 marketplace; instead, it describes a flow where a user announces a desired output, willingness to pay, and service providers compete to fulfill the job requirement in the best way possible.
+
+### Actors
+There are two actors in the workflow described in this NIP:
+* Customers (npubs who request a job)
+* Service providers (npubs who fulfill jobs)
+
+## Job request (`kind:5000-5999`)
+A request to process data, published by a customer. This event signals that an customer is interested in receiving the result of some kind of compute.
+
+```json
+{
+    "kind": 5xxx, // kind in 5000-5999 range
+    "content": "",
+    "tags": [
+        [ "i", "<data>", "<input-type>", "<relay>", "<marker>" ],
+        [ "output", "<mime-type>" ],
+        [ "relays", "wss://..."],
+        [ "bid", "<msat-amount>" ],
+        [ "t", "bitcoin" ]
+    ]
+}
+```
+
+All tags are optional.
+
+* `i` tag: Input data for the job (zero or more inputs)
+    * `<data>`: The argument for the input
+    * `<input-type>`: The way this argument should be interpreted. MUST be one of:
+        * `url`: A URL to be fetched of the data that should be processed.
+        * `event`: A Nostr event ID.
+        * `job`: The output of a previous job with the specified event ID. The dermination of which output to build upon is up to the service provider to decide (e.g. waiting for a signaling from the customer, waiting for a payment, etc.)
+        * `text`: `<data>` is the value of the input, no resolution is needed
+    * `<relay>`: If `event` or `job` input-type, the relay where the event/job was published, otherwise optional or empty string
+    * `<marker>`: An optional field indicating how this input should be used within the context of the job
```

Comment:
There is a potential for confusing the relay as the marker or the marker as the relay, given that they are both optional.

How about making it mandatory that
- `<relay>` MUST be present for `event` or `job` input-type, 
- `<relay>` MUST be absent for all other types that don't require querying relays

Reply:
this is already convention per other NIPs that do the same, when wanting to leave relay empty and put a marker the relay should be an empty string. I can add a note to make this explicit here.